lib/dor/text_extraction/ocr.rb

```diff
@@ -20,6 +20,20 @@ def abbyy_input_path
         File.join(Settings.sdr.abbyy.local_ticket_path, bare_druid)
       end
 
+      def cleanup
+        if Dir.exist?(abbyy_input_path)
+          raise "#{abbyy_input_path} is not empty" unless Dir.empty?(abbyy_input_path)
+
+          FileUtils.rm_rf(abbyy_input_path)
+        end
+
+        return true unless Dir.exist?(abbyy_output_path)
+
+        raise "#{abbyy_output_path} is not empty" unless Dir.empty?(abbyy_output_path)
+
+        FileUtils.rm_rf(abbyy_output_path)
+      end
+
```

Comment:
this implementation:

- does not cause errors if the folders don't exist (it will just continue)
- will raise an error if the folder exists but is not empty (this is not expected)
- if the input folder is empty, but the output folder is not empty, it will delete the input folder first and then raise an exception on the output folder not being empty ... it is fine to re-run later and it will cleanup the output folder (since non-existence of a folder is not an issue)

Reply:
two questions:
1. is the ticket in the output path?  if not, is gating the ticket file deletion on the existence of the output path intentional?
2. if the input dir is not empty, do we want to refrain from deleting the output dir?

i guess the overall gist of what i'm wondering is, do we want to condition any deletions on other deletions, or do we just want to clean up everything we can safely clean up independently of the contents of paths that can't yet be removed?